DigitalLearningSolutions.Data.Migrations/202204071530_PopulateUsersTableFromAccountsTables.cs

```diff
@@ -0,0 +1,415 @@
+                        var existingUserWithEmail = connection.QuerySingleOrDefault<User>(
+                            "SELECT * FROM Users WHERE PrimaryEmail = @email",
+                            new { email = emailGroup.Key }
+                        );
+
+                        if (existingUserWithEmail != null)
+                        {
+                            // If we find a user, we update any default data on that user
+                            UpdateExistingUserWithDelegateDetails(
+                                connection,
+                                existingUserWithEmail,
+                                delegateAccount,
+                                allJobGroupsMatch
+                            );
+                        }
+                        else
+                        {
+                            // Otherwise we create a new user with the email address
+                            InsertNewUserForDelegateAccount(connection, delegateAccount, false);
+                        }
+                    }
+                }
+                else
+                {
+                    // If we don't have a valid email we create new users for each with a guid email
+                    foreach (var delegateAccount in emailGroup)
+                    {
+                        InsertNewUserForDelegateAccount(connection, delegateAccount, true);
+                    }
+                }
+            }
+
+            transactionScope.Complete();
+        }
+
+        public override void Down()
+        {
+            Execute.Sql(Resources.UAR_859_PopulateUsersTableFromAccountsTables_DOWN);
+        }
+
+        private static void UpdateExistingUserWithDelegateDetails(
+            IDbConnection connection,
+            User existingUserWithEmail,
+            DelegateAccount delegateAccount,
+            bool allJobGroupsMatch
+        )
+        {
+            connection.Execute(
+                @"UPDATE Users
+                            SET
+                                PasswordHash = @passwordHash,
+                                FirstName = @firstName,
+                                LastName = @lastName,
+                                JobGroupID = @jobGroupId,
+                                ProfessionalRegistrationNumber = @professionalRegistrationNumber,
+                                ProfileImage = @profileImage,
+                                Active = @active,
+                                ResetPasswordID = @resetPasswordId,
+                                HasBeenPromptedForPrn = @hasBeenPromptedForPrn,
+                                LearningHubAuthId = @learningHubAuthId,
+                                HasDismissedLhLoginWarning = @hasDismissedLhLoginWarning,
+                                EmailVerified = GETUTCDATE(),
+                                DetailsLastChecked = CASE WHEN @detailsMatched = 0 OR DetailsLastChecked IS NULL THEN NULL ELSE GETUTCDATE()

```

Comment:
Looks like there should technically be an END on here. Or you can use IIF if you prefer the look of it, but I'm happy with either.